### ISSUE TITLE
Set up alarms for when our Functions fail

### ISSUE THREAD

**Issue by maximumpeaches** [Nonlinear-EA/The-Nonlinear-Library#38]
See the context here #36 as well as some steps that have already been taken.

**Comment by maximumpeaches** [comment]
We may also want to do some XML validation and log / throw specific errors when it's invalid.

**Comment by aarangop** [comment]
from #36 
>Hey I'm a little confused why we need a separate function to check the XML
files. Won't the Functions we already have fail and log the error.

Somehow the files are still updated even when invalid xml is created. With the new code this shouldn't be necessary and we could log an error when the xml is invalid inside each function. However, none of the current functions do proper xml parsing, so we might still get invalid xml. Since the current functions are hardcoded on the GCP Functions or spread across multiple files on the repo, adding this check to each one would be a cumbersome, so, I think it's way easier to write this function to check the xml's integrity.

I deployed an xml integrity check Function `do_xml_integrity_checks`. The function logs a CRITICAL message when an xml is invalid and I set up an alert on the Logs Explorer that should notify us when a log's payload matches "^CRITICAL" (regex). Don't know yet how to log stuff that is detected as CRITICAL or ERROR by GCP. This way we'll at least know when an xml file is invalid and we won't have to wait for users to notify us.

**Comment by maximumpeaches** [comment]
Yes makes sense. It's also okay to wait to worry about this stuff until
after you're done refactoring as it'll be easier to fix then (just one file
to fix instead of a bunch, as you mentioned). The vast majority of our
users just listen to The Nonlinear Library because Kat hasn't announced the
other podcasts anywhere yet - you can see our usage stats on the Spotify
for Podcasters page for instance. Up to you what you do but just letting
you know about this in case it's helpful at some point.
